src/plugins/kibana_react/public/theme/kibana_theme_provider.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useMemo } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { Observable } from 'rxjs';
+import { EuiThemeProvider } from '@elastic/eui';
+import type { CoreTheme } from '../../../../core/public';
+import { getColorMode } from './utils';
+
+interface KibanaThemeProviderProps {
+  theme$: Observable<CoreTheme>;
+}
+
+const defaultTheme: CoreTheme = {
+  darkMode: false,
+};
+
+export const KibanaThemeProvider: FC<KibanaThemeProviderProps> = ({ theme$, children }) => {
+  const theme = useObservable(theme$, defaultTheme);
+  const colorMode = useMemo(() => getColorMode(theme), [theme]);
+  return <EuiThemeProvider colorMode={colorMode}>{children}</EuiThemeProvider>;
+};
```

Comment:
So, this is the 'public' theme provider exposed to plugins via `kibana_react`.

At the moment, it's a plain wrapper around `EuiThemeProvider`.

I asked myself if this was alright to just expose's eui theme, or if we should also use it to expose our own context containing the current `CoreTheme`.

e.g

```tsx
const Component = () => {
   const coreTheme = useCoreTheme();
   // ...
}

// ...

<KibanaThemeProvider theme$={theme$}>
   <Component />
</KibanaThemeProvider>
```

As the addition is fairly trivial, I came to the conclusion that we should probably wait until the other teams actually use the provider to see what exact need for further tooling they have, wdyt?

Reply:
this makes sense to me, I can see how this can be added without requiring refactors for consumers.